contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,146 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_json_query' LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
+-- Function to convert the path format into expected jsonb_set path format using Regular Expression
+CREATE OR REPLACE FUNCTION sys.convert_jsonpath_to_jsonb_path(json_path TEXT)
+RETURNS TEXT
+AS
+$BODY$
+BEGIN
+    SELECT regexp_replace(json_path, '\$\.|]|\$\[' , '' , 'ig') INTO json_path; -- To remove the "$." and "]" sign from the string 
+    SELECT regexp_replace(json_path, '\.|\[' , ',' , 'ig') INTO json_path; -- To replace the "." and "[" with the "," to change into required format
+    SELECT CONCAT('{',json_path,'}') INTO json_path; -- Final required format of path by jsonb_set
+    return json_path;
+END
+$BODY$
+LANGUAGE plpgsql;
+
+--Function to set the append_modifier and create_if_missing flags
+CREATE TYPE sys.json_modify_flags AS (create_if_missing BOOL, append_modifier BOOL);
+
+CREATE OR REPLACE FUNCTION sys.set_json_modify_flags(path_json TEXT)
+RETURNS json_modify_flags
+AS
+$BODY$
+DECLARE
+    result_mode_flags json_modify_flags;
+    word_count INTEGER;
+    create_if_missing BOOL:='TRUE';
+    mode TEXT;
+    modifier TEXT;
+    append_modifier BOOL:='FALSE';
+BEGIN
+    
+    word_count = LENGTH(path_json) - LENGTH(REPLACE(path_json, ' ', '')) + 1; --To calculate the number of words in the path_json
```

Comment:
Yes, you are right. It will be better to use regexp_split_to_array() instead of this. I will make the changes and implement it using regexp_split_to_array().